lib/class.OS_Linux.php

```diff
@@ -935,18 +935,19 @@ public function getBattery() {
 		// Get vals for each battery
 		foreach ($bats as $b) {
 
-			$go_for_it = true;
-
-			foreach(array($b.'/manufacturer', $b.'/status', $b.'/charge_now') as $f)
-				if (!is_file($f))
-					$go_for_it = false; // Continue out of two nested loops
-
-			if (!$go_for_it) 
-				continue;
+			foreach(array($b.'/manufacturer', $b.'/status') as $f)
+				if (!is_file($f))continue;
 
 			// Get these from the simple text files
-			$charge_full = LinfoCommon::getIntFromFile($b.'/charge_full');
-			$charge_now = LinfoCommon::getIntFromFile($b.'/charge_now');
+			if(is_file($b.'/charge_full'))
+				{
+				$charge_full = LinfoCommon::getIntFromFile($b.'/charge_full');
+				$charge_now = LinfoCommon::getIntFromFile($b.'/charge_now');
+				}
+			else{
+				$charge_full = LinfoCommon::getIntFromFile($b.'/energy_full');
+				$charge_now = LinfoCommon::getIntFromFile($b.'/energy_now');
+				}
 
```

Comment:
Please mind the braces and indention. Keep a space between the else and { and keep the indention levels consistent.